### ISSUE TITLE
Different results running on CPU and GPU

### ISSUE THREAD

**Issue by yujianll** [IDEA-Research/GroundingDINO#65]
Hi, thanks for releasing the repo.

I encountered a problem where I get different results for the same input when running on CPU and GPU. For example, I get the following results using CPU and GPU. I wonder what might cause the issue.

![cpu](https://user-images.githubusercontent.com/46540151/233474162-a9b5f63a-6f90-4551-95ff-de5d8c093951.jpg)
![gpu](https://user-images.githubusercontent.com/46540151/233474166-a3b9aee5-73f5-430e-a00e-36cf6c283944.jpg)

**Comment by yujianll** [comment]
Also, the output logits has shape [bs, 900, 256]. Based on the explanation, does 256 refers to number of tokens (padded to 256)? Thanks!

**Comment by lfxx** [comment]
i meet this too.Seems like the performance on CPU is always better than GPU

**Comment by yujianll** [comment]
@lfxx I wonder if you find a solution for this? I still can't detect anything on GPU.

**Comment by lfxx** [comment]
> @lfxx I wonder if you find a solution for this? I still can't detect anything on GPU.

I can detect object on GPU,but is worse than using CPU.If you can not detect anything on GPU,maybe there is something else cause this.

**Comment by darshats** [comment]
I have the same issue, the cpu variant finds lesser objects for me than the gpu one. Using default threshold for box and class.
